### ISSUE TITLE
Ti.Admob crashes on launch of the application.

### ISSUE THREAD

**Issue by curtx** [tidev/ti.admob#306]
I tried 2.5.1 and 3.0.0 for IOS using titanium sdk 10.1.1.GA. The app launches and then crashes at start screen. After debugging I found this error

 pthread_exit() called from a thread not created by pthread_create()

Please help!

**Comment by deckameron** [comment]
Hi @curtx ,

Try using [this one](https://github.com/Astrovic/ti.admob/releases/tag/ios-4.8.0) and see it solves the issue for you.
Best of luck.

**Comment by curtx** [comment]
Hi @deckameron ,

Thanks for the reply. I tried the 4.8.0 version on both simulator and iPhone 7, but still app is crashing at launch.

**Comment by Manuelapr** [comment]
Hi, have you figured it out? I'm having the same problem and have tried the admob version 4.0.0 and the 4.8.0.

**Comment by curtx** [comment]
@Manuelapr Yes, the issue can be resolved by setting up Info. list correctly. Once the Info. plist is set up correctly with keys mentioned in the docs

**Comment by Manuelapr** [comment]
Thanks for you replay. Can you guide me on how to set up the info.plist file?

Thanks

**Comment by curtx** [comment]
Here use this document to set keys correctly
https://developers.google.com/ad-manager/mobile-ads-sdk/ios/quick-start#update_your_infoplist

**Comment by Manuelapr** [comment]
Hi, thanks for the replay.

Using that link I managed to make it to work. At least the app don't closes any mor 😁

But now I get this message:

Failed to read publisher's account configuration; try again later.


I have used the example code that came with the module and have set in all places the keys of my account and ad.


What could I bee missing with this error?

**Comment by curtx** [comment]
Yes, now you need to set up your own admob account and provide proper settings especially Admob UnitID

**Comment by Manuelapr** [comment]
Hi @curtx . Thanks for the replay. I have tried to set the id's in this places shown in the code of the example that cames with the module.

I have also posted the ID's on mu admob account.

My problem is that I get allways the message:


"Failed to read publisher's account configuration; try again later."


on the admob.requestConsentInfoUpdateWithParameters method.

I really don't know what iI'm missing.


`var bannerAdView = Admob.createView({
    height: 50,
    bottom: 0,
    adType: Admob.AD_TYPE_BANNER,
    adUnitId: 'ca-app-pub-1867145378940081~541672XXXX', 
    adBackgroundColor: 'black',
    contentURL: 'https://mywebsite', 
    requestAgent: 'AppName',
    extras: {
      'version': 1.0,
      'name': 'AppName'
    }, 
    tagForChildDirectedTreatment: false,
    keywords: ['Aquarium', 'fish']
  });
  win.add(bannerAdView);


var ad2 = Admob.createView({
      adType: Admob.AD_TYPE_INTERSTITIAL,
      adUnitId: 'ca-app-pub-1867145378940081~541672XXXX', 
      keywords: ['Aquarium', 'fish']
    });


var rewardedVideo = Admob.createView({
      adUnitId: 'ca-app-pub-1867145378940081~541672XXXX',
      adType: Admob.AD_TYPE_REWARDED_VIDEO,
    });


Admob.requestConsentInfoUpdateWithParameters({
    publisherIdentifiers: "pub-186714537894XXXX",
    tagForUnderAgeOfConsent: false,
    callback: function (e) {
`


Publisher ID: pub-186714537894XXXX

APP ID: ca-app-pub-1867145378940081~541672XXXX

Advertisment blok id: ca-app-pub-1867145378940081/509329XXXX

**Comment by Astrovic** [comment]
You are confusing **AdMob ad Unit ID** with **AdMob app ID** (https://support.google.com/admob/answer/7356431).

AdMob ad Unit ID is the one in this format `ca-app-pub-xxxxxxxxxxxxxxxx~xxxxxxxxxx` and you must enter it only in the `tiapp.xml` in the `GADApplicationIdentifier` key.
```xml
<ios>
    <plist>
        <dict>
            <key>GADApplicationIdentifier</key>
            <string>ca-app-pub-xxxxxxxxxxxxxxxx~xxxxxxxxxx</string> 
        </dict>
    </plist>
</ios>
```

The AdMob ad Unit ID is the ad that you create in your AdMob account for your app, and it is in the format `ca-app-pub-xxxxxxxxxxxxxxxx/xxxxxxxxxx` and you must insert it in 
```js
Admob.createView({
   adUnitId: 'app-pub-xxxxxxxxxxxxxxxx/xxxxxxxxxx'
});
```

For example, to create a banner in your AdMob account you must follow these instructions: https://support.google.com/admob/answer/7311346
